src/highlight.js

```diff
@@ -131,25 +130,45 @@ function wrapLinesInSpan(codeTree, lineStyle) {
   return newTree;
 }
 
+function defaultRenderer({ rows, style, useInlineStyles }) {
+  return (
+    rows.map((node, i) => createElement({
+      node,
+      style,
+      useInlineStyles,
+      key: `code-segement${i}`
+    }))
+  );
+}
+
 export default function (lowlight, defaultStyle) {
- return function SyntaxHighlighter(props) {
-    const {
-      language,
-      children,
-      style = defaultStyle,
-      customStyle = {},
-      codeTagProps = {},
-      useInlineStyles = true,
-      showLineNumbers = false,
-      startingLineNumber = 1,
-      lineNumberContainerStyle,
-      lineNumberStyle,
-      wrapLines = false,
-      lineStyle = {},
-      ...rest
-    } = props;
-    const codeTree = language ? lowlight.highlight(language, children) : lowlight.highlightAuto(children);
-    const defaultPreStyle = style.hljs || {backgroundColor: '#fff'};
+ return function SyntaxHighlighter({
+  language,
+  children,
+  style = defaultStyle,
+  customStyle = {},
+  codeTagProps = {},
+  useInlineStyles = true,
+  showLineNumbers = false,
+  startingLineNumber = 1,
+  lineNumberContainerStyle,
+  lineNumberStyle,
+  wrapLines = false,
+  lineStyle = {},
+  renderer,
+  ...rest
+ }) {
+    /* custom renderers rely on individual row elements so we need to turn wrapLines on 
+     * if renderer is provided
+    */
+    wrapLines = renderer ? true : wrapLines;
```

Comment:
not sure if all custom renderers would but virtualization would, it might be confusing if you forget this prop? maybe wrapLines should be on by default?